File: app/src/main/java/com/example/bettersleep/entity/Sleep.java
package com.example.bettersleep.entity;

import androidx.annotation.NonNull;
import androidx.room.ColumnInfo;
import androidx.room.Entity;
import androidx.room.PrimaryKey;

@Entity(tableName = "sleep_table")
public class Sleep {
    @PrimaryKey(autoGenerate = true)
    @NonNull
    @ColumnInfo(name = "id")
    private final Integer id = 0;

    @NonNull
    @ColumnInfo(name = "date")
    private final String date;

    @NonNull
    @ColumnInfo(name = "time")
    private final String time;

    @NonNull
    @ColumnInfo(name = "duration")
    private final Integer duration;


    public Sleep(@NonNull String date, @NonNull String time, @NonNull Integer duration) {
        this.date = date;
        this.time = time;
        this.duration = duration;
    }

    @NonNull
    public Integer getId(){
        return this.id;
    }

    @NonNull
    public String getDate(){
        return this.date;
    }

    @NonNull
    public String getTime(){
        return this.time;
    }

    @NonNull
    public Integer getDuration(){return this.duration;}


}

File: app/src/main/java/com/nursyah/bettersleep/entity/Sleep.kt
package com.nursyah.bettersleep.entity

import androidx.room.PrimaryKey
import androidx.room.ColumnInfo
import androidx.room.Entity

@Entity(tableName = "sleep_table")
data class Sleep(
    @field:ColumnInfo(name = "id")
    @field:PrimaryKey(autoGenerate = true) val id:Long = 0,

    @field:ColumnInfo(name = "date") val date:String,

    @field:ColumnInfo(name = "duration") val duration:Int,
)